src/field_player_avatar.c

```diff
@@ -2127,22 +2130,56 @@ static bool32 Fishing_DoesFirstMonInPartyHaveSuctionCupsOrStickyHold(void)
     return (ability == ABILITY_SUCTION_CUPS || ability == ABILITY_STICKY_HOLD);
 }
 
-static bool32 Fishing_RollForBite(bool32 isStickyHold)
+static bool32 Fishing_RollForBite(u32 rod, bool32 isStickyHold)
 {
-    return ((Random() % 100) > CalculateFishingBiteOdds(isStickyHold));
+    return ((Random() % 100) > CalculateFishingBiteOdds(rod, isStickyHold));
 }
 
-static u32 CalculateFishingBiteOdds(bool32 isStickyHold)
+static u32 CalculateFishingBiteOdds(u32 rod, bool32 isStickyHold)
 {
-    u32 odds = FISHING_DEFAULT_ODDS;
+    u32 odds;
+
+    if (rod == OLD_ROD)
+        odds = FISHING_OLD_ROD_ODDS;
+    if (rod == GOOD_ROD)
+        odds = FISHING_GOOD_ROD_ODDS;
+    if (rod == SUPER_ROD)
+        odds = FISHING_SUPER_ROD_ODDS;
+
+    odds -= CalculateFishingFollowerBoost();
 
     if (isStickyHold)
-        odds -= FISHING_STICKY_BOOST;
+    {
+        if (I_FISHING_STICKY_BOOST >= GEN_4)
+            odds -= (100 - odds);
+        else
+            odds -= FISHING_STICKY_BOOST;
+    }
 
     odds -= CalculateFishingProximityBoost(odds);
+
     return odds;
 }
 
+static u32 CalculateFishingFollowerBoost()
+{
+    struct Pokemon *mon = GetFirstLiveMon();
+
+    if (!I_FISHING_FOLLOWER_BOOST || !mon)
+        return 0;
+
+    if (GetMonData(mon, MON_DATA_FRIENDSHIP) >= 250)
+        return 50;
+    else if (GetMonData(mon, MON_DATA_FRIENDSHIP) >= 200)
+        return 40;
+    else if (GetMonData(mon, MON_DATA_FRIENDSHIP) >= 150)
+        return 30;
+    else if (GetMonData(mon, MON_DATA_FRIENDSHIP) >= 100)
```

Comment:
All these ``GetMonData`` checks are eating up an unnecessary amount of cycles. Fetch the friendship level once, store it in a local variable, and check the variable